File: app/src/main/java/com/demo/customview/sort/PinyinComparator.java
package com.demo.customview.sort;

import android.text.TextUtils;

import java.util.Comparator;

/**
 * created by 姚明亮
 * Time：2019/8/20 14:04
 */
public class PinyinComparator implements Comparator<SortBean> {
    @Override
    public int compare(SortBean sortBean, SortBean t1) {
        if (TextUtils.equals(sortBean.letter, "#")) {
            return 1;
        } else if (TextUtils.equals(t1.letter, "#")) {
            return -1;
        } else {
            return sortBean.pinyin.compareTo(t1.pinyin);
        }
    }
}

File: app/src/main/java/com/demo/customview/sort/PinyinComparator.kt
package com.demo.customview.sort

import android.text.TextUtils
import java.util.*

/**
 * Author by YML, Date on 2021/3/9.
 * PS: Not easy to write code, please indicate.
 */
class PinyinComparator: Comparator<SortBean> {
    override fun compare(sortBean: SortBean, t1: SortBean): Int {
        return when {
            TextUtils.equals(sortBean.letter, "#") -> {
                1
            }
            TextUtils.equals(t1.letter, "#") -> {
                -1
            }
            else -> {
                sortBean.pinyin.compareTo(t1.pinyin)
            }
        }
    }
}